modules/external-dns-iam/variables.tf

```diff
@@ -1,6 +1,6 @@
 variable "name_prefix" {
   description = "Name to prefix the resources with."
-  default     = ""
+  default     = "example"
```

Comment:
I would recommend we just drop the default completely and let that be a required parameter.